common/src/main/java/io/pravega/common/concurrent/Futures.java

```diff
@@ -44,6 +45,31 @@
  */
 public final class Futures {
 
+    /**
+     * Makes sure that the result future given out to caller is actually completed on caller's executor.
+     * So any chaining, if done without specifying an executor, will either happen on
+     * caller's executor or fork join pool but never on someone else's executor.
+     *
+     * @param future The future to execute.
+     * @param <T> The Type of the future's result.
+     * @param executor The executor for executing the future.
+     * @return <T>CompletableFuture<T> Result of execution, successful or exceptional.
```

Comment:
`A new {@link CompletableFuture} that will complete with the same outcome as the given one, but on the given {@link Executor}.`